cdap-unit-test/src/test/java/co/cask/cdap/security/AuthorizationTest.java

```diff
@@ -1028,6 +1034,82 @@ public void testCrossNSSpark() throws Exception {
     appManager.stopAll();
   }
 
+  @Test
+  public void testScheduleAuth() throws Exception {
+    createAuthNamespace();
+    ApplicationManager appManager = deployApplication(AUTH_NAMESPACE.toId(), AppWithSchedule.class);
+    ProgramId workflowID = new ProgramId(AUTH_NAMESPACE.getNamespace(), AppWithSchedule.class.getSimpleName(),
+                                         ProgramType.WORKFLOW, AppWithSchedule.SampleWorkflow.class.getSimpleName());
+
+    final WorkflowManager workflowManager =
+      appManager.getWorkflowManager(AppWithSchedule.SampleWorkflow.class.getSimpleName());
+    ScheduleManager scheduleManager = workflowManager.getSchedule(AppWithSchedule.SCHEDULE_NAME);
+
+    // switch to BOB
+    SecurityRequestContext.setUserId(BOB.getName());
+    // try to resume schedule as BOB. It should fail since BOB does not have privileges on the programs
+    try {
+      scheduleManager.resume();
+      Assert.fail("Resuming schedule should have failed since BOB does not have EXECUTE on the program");
+    } catch (Exception e) {
```

Comment:
Can we just catch `UnAuthorizedException` here? Then we will not need the assert below.